apps/api/src/services/external/apiNodeService.ts

```diff
@@ -32,41 +34,86 @@ import { getDeploymentRelatedMessages } from "../db/deploymentService";
 import { getProviderList } from "../db/providerStatusService";
 
 export async function getChainStats() {
+  const createLoggingExecutor = (logger: LoggerService) => async <T>(cb: () => Promise<T>, defaultValue?: T): Promise<T> => {
```

Comment:
what I meant is to extract this function into a dedicated module/file/service and then use it here and in another file